crates/uv-auth/src/middleware.rs

```diff
@@ -56,8 +105,10 @@ pub struct AuthMiddleware {
     netrc: NetrcMode,
     keyring: Option<KeyringProvider>,
     cache: Option<CredentialsCache>,
-    /// We know that the endpoint needs authentication, so we don't try to send an unauthenticated
-    /// request, avoiding cloning an uncloneable request.
+    /// Auth policies for specific URLs.
+    url_auth_policies: UrlAuthPolicies,
+    /// Set all endpoints as needing authentication. We never try to send an
+    /// unauthenticated request, avoiding cloning an uncloneable request.
     only_authenticated: bool,
```

Comment:
Can we remove this entirely in favor of `url_auth_policies`? (could be a follow-up refactor)

Reply:
Yeah but I don't think we need `only_authenticated` long-term, we can just set an auth policy for the publish URL (which is the one place it's used)